docs/internals/optimizer.rst

```diff
@@ -1121,6 +1122,46 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes ``sstore`` and memory store statements if conditions are met for said removal.
```

Comment:
```suggestion
Optimizer component that removes redundant ``sstore`` and memory store statements.
```

``if conditions are met`` is rather redundant itself, isn't it? It of course doesn't to it unconditionally for all of them :-).

Reply:
We could also phrase it more abstractly as something like ``removes statements with side-effects on storage or memory in cases in which it can prove that those side-effects can never be observed``... although I'm not sure about that - might make things worse...

Reply:
Strictly speaking that'd be more accurate, though... one example of a weird side-effect of memory is that ``msize()`` can change if you remove memory stores, so the step actually also doesn't remove any memory store, if there is any ``msize()`` in the code... maybe we don't need to mention that detail specifically here, but yeah.

Reply:
I honestly prefer the first suggestion as `redundant` is explicit, and pretty simple to decipher for someone reading the docs, where as `statements with side-effects`, and `side-effects can never be observed` is a little too abstract for my tastes :)